main_test.go

```diff
@@ -53,21 +42,169 @@ func TestCreateCustomer_withAMissingEmailArg_returnsAnError(t *testing.T) {
+	db, err := connectDB()
+	if err != nil {
+		t.Fatal(err)
 	}
 
-	w := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.AlignRight|tabwriter.Debug)
-	c.Print(w)
-	w.Flush()
-	// Output:
-	// 1| zack.patrick@outreach.io|WA
+	app := &cli.App{
+		Commands: []*cli.Command{
+			newCreateCustomerCommand(db),
+		},
+	}
+	err = app.Run([]string{"store", "create-customer", "vivek.shah@outreach.io", "PP"})
+	assert.ErrorContains(t, err, "State must be a valid U.S. State or Territory")
 }
+
+func TestCreateCustomer_withANonTwoLetterStateCode_returnsAnError(t *testing.T) {
+	db, err := connectDB()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	app := &cli.App{
+		Commands: []*cli.Command{
+			newCreateCustomerCommand(db),
+		},
+	}
+	err = app.Run([]string{"store", "create-customer", "vivek.shah@outreach.io", "PPP"})
+	assert.ErrorContains(t, err, "State length must be 2")
+}
+
+func TestCreateCustomer_withValidInput_entersDatabaseCorrectly(t *testing.T) {
+	db, err := connectDB()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	app := &cli.App{
+		Commands: []*cli.Command{
+			newCreateCustomerCommand(db),
+		},
+	}
+	_, err = db.Exec("DELETE FROM Customers")
+	assert.NilError(t, err)
+	_, err = db.Exec("ALTER TABLE Customers AUTO_INCREMENT=1")
+	assert.NilError(t, err)
+	err = app.Run([]string{"store", "create-customer", "vivek.s@outreach.io", "WA"})
+	assert.NilError(t, err)
+
+	rows := db.QueryRow("SELECT * FROM Customers ORDER BY ID DESC LIMIT 1")
+
+	assert.NilError(t, rows.Err())
+	var id int
+	var email string
+	var state string
+	rows.Scan(&id, &email, &state)
+	assert.Equal(t, email, "vivek.s@outreach.io")
+	assert.Equal(t, state, "WA")
+	assert.Equal(t, id, 1)
+
+}
+
+func TestCreateMultipleCustomer_withValidInput_entersDatabaseCorrectly(t *testing.T) {
+	db, err := connectDB()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	app := &cli.App{
+		Commands: []*cli.Command{
+			newCreateCustomerCommand(db),
+		},
+	}
+	_, err = db.Exec("DELETE FROM Customers")
+	assert.NilError(t, err)
+	_, err = db.Exec("ALTER TABLE Customers AUTO_INCREMENT=1")
+	assert.NilError(t, err)
+	err = app.Run([]string{"store", "create-customer", "vivek.s@outreach.io", "WA"})
+	assert.NilError(t, err)
+
+	err = app.Run([]string{"store", "create-customer", "v.s@outreach.io", "CA"})
+	assert.NilError(t, err)
+
+	rows := db.QueryRow("SELECT * FROM Customers ORDER BY ID DESC LIMIT 1")
+
+	assert.NilError(t, rows.Err())
+	var id int
+	var email string
+	var state string
+	rows.Scan(&id, &email, &state)
+	assert.Equal(t, email, "v.s@outreach.io")
+	assert.Equal(t, state, "CA")
+	assert.Equal(t, id, 2)
+
+}
+
+func Example_CreateCustomer_hasCorrectPrintOutput() {
+	db, _ := connectDB()
+
+	app := &cli.App{
+		Commands: []*cli.Command{
+			newCreateCustomerCommand(db),
+		},
+	}
+	db.Exec("DELETE FROM Customers")
+
+	db.Exec("ALTER TABLE Customers AUTO_INCREMENT=1")
+
+	app.Run([]string{"store", "create-customer", "vivek.s@outreach.io", "WA"})
```

Comment:
Seeing this "clear and reset the database" logic repeated in quite a few spots - a couple of thoughts there:
1. It would probably make sense to move that to a helper function each test can call:
```
func setupTestDB(t *testing.T) *sql.DB {
  db := ...
	_, err = db.Exec("DELETE FROM Customers")
	assert.NilError(t, err)
	_, err = db.Exec("ALTER TABLE Customers AUTO_INCREMENT=1")
	assert.NilError(t, err)
	
	return db
}

func TestA(t *testing.T) {
  db := setupTestDB(t)
  ...
}

func TestB(t *testing.T) {
  db := setupTestDB(t)
  ...
}
```

The second thought is a bit more philosophical:  the logic dealing with printing a customer doesn't really have anything to do with the database (or, perhaps better put: we can design our code in a where where the customer printing logic and database logic are completely independent of each other). 

Think about it this way: if I have a customer with id=1,email="vivek.shah@outreach.io",state="WA", we know how it should get printed - that can be tested independent of where that data is loaded from. If we structured this as something like:
```
row := selectCustomerFromDB()
customer := convertRowToCustomer(row)
printCustomer(customer)
```

Then it becomes much easier to test the `printCustomer` function since that logic is no longer tied to anything related to the database:
```go
func TestPrintCustomer(t *testing.T) {
  customer := Customer{
    ID: 1,
    Email: "vivek.shah@outreach.io",
    State: "WA",
  }
  
  printCustomer(customer)
  // Output:
  // ...
}